rero_ils/modules/patron_transactions/api.py

```diff
@@ -103,156 +106,57 @@ def update(self, data, commit=True, dbcommit=True, reindex=True):
-        :return: return total amount of transactions.
-        """
-        search = cls._build_transaction_query(patron_pid, status, types)
-        if not with_subscription:
-            search = search.exclude('terms', type=['subscription'])
-        search.aggs.metric('transactions_total_amount',
-                           'sum', field='total_amount')
-        # set the from/size to 0
-        search = search[0:0]
-        results = search.execute()
-        return results.aggregations.transactions_total_amount.value
-
-    @classmethod
-    def get_transactions_count_for_patron(cls, patron_pid, status=None):
-        """Get patron transactions count linked to a patron.
-
-        :param patron_pid: the patron pid being searched
-        :param status: (optional) transaction status filter,
-        """
-        query = cls._build_transaction_query(patron_pid, status)
-        return query.source().count()
-
-    @classmethod
-    def get_transactions_pids_for_patron(cls, patron_pid, status=None):
-        """Get patron transactions pids linked to a patron.
-
-        :param patron_pid: the patron pid being searched
-        :param status: (optional) transaction status filter,
-        """
-        query = cls._build_transaction_query(patron_pid, status)
-        return sorted_pids(query)
-
-    @classmethod
-    def get_transactions_by_patron_pid(cls, patron_pid, status=None):
-        """Return opened fees for patron.
+    # GETTER & SETTER =========================================================
+    #   * Define some properties as shortcut to quickly access object attrs.
+    #   * Defines some getter methods to access specific object values.
 
-        :param patron_pid: the patron PID
-        :param status: the status of transaction
-        :return: return an array of transaction
-        """
-        query = cls._build_transaction_query(patron_pid, status)
-        return query\
-            .params(preserve_order=True)\
-            .sort({'creation_date': {'order': 'asc'}})\
-            .scan()
-
-    @classmethod
-    def get_last_transaction_by_loan_pid(cls, loan_pid, status=None):
-        """Return last fee for loan.
+    @property
+    def loan_pid(self):
+        """Get the `Loan` pid related to this transaction."""
+        if self.get('loan'):
+            return extracted_data_from_ref(self['loan'])
 
-        :param loan_pid: the loan PID
-        :param status: the status of transaction
-        :return: return last transaction transaction
-        """
-        query = PatronTransactionsSearch() \
-            .filter('term', loan__pid=loan_pid)
-        if status:
-            query = query.filter('term', status=status)
-        results = query\
-            .sort({'creation_date': {'order': 'desc'}})\
-            .source('pid').scan()
-        try:
-            pid = next(results).pid
-            return PatronTransaction.get_record_by_pid(pid)
-        except StopIteration:
-            pass
+    @cached_property
```

Comment:
It seems reasonable to use `cached_property` here, because theses 2 priorities are often use and "loan" and "notification" will never change during a PatronTransaction life cycle.